src/mgos_imu_icm20948.c

```diff
@@ -0,0 +1,482 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mgos.h"
+#include "mgos_i2c.h"
+#include "mgos_imu_icm20948.h"
+
+#define ACC_BASE_ODR  1125.f
+#define GYRO_BASE_ODR 1100.f
+
+static bool mgos_imu_icm20948_change_bank(struct mgos_i2c *i2c, uint8_t i2caddr, void *imu_user_data, uint8_t bank_no) {
+  struct  mgos_imu_icm20948_userdata *iud = (struct mgos_imu_icm20948_userdata *)imu_user_data;
+  uint8_t bank_addr = 0x00;
+
+  if(bank_no == iud->current_bank_no) {
```

Comment:
Suggest you check for iud to exist before accessing it.
if (!imu_user_data) return false;